File: 2154-minimum-moves-to-convert-string/minimum-moves-to-convert-string.java
class Solution {
    public int minimumMoves(String s) {
        int moves = 0;
        int idx = 0, size = s.length();

        while (idx < size) {
            if (s.charAt(idx) == 'X') {
                moves++;
                idx += 3;
            } else {
                idx += 1;
            }
        }

        return moves;
    }
}

File: 2154-minimum-moves-to-convert-string/minimum-moves-to-convert-string.py
class Solution:
    def minimumMoves(self, string: str) -> int:
        moves: int = 0
        if string:
            idx, size = 0, len(string)
            while idx < size:
                if string[idx] == 'X':
                    moves += 1
                    idx += 3
                else:
                    idx += 1
        
        return moves